drivers/pinmux/pinmux_cc2650.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * License: Apache-2.0
+ *
+ * Driver for the I/O controller (pinmux) for
+ * Texas Instrument's CC2650 SoC.
+ *
+ * For these SoCs, available pin functions are as follows:
+ *
+ * 0x00 - GPIO
+ * 0x07 - AON 32 Khz clock
+ * 0x08 - AUX IO
+ * 0x09 - SSI0 RX
+ * 0x0A - SSI0 TX
+ * 0x0B - SSI0 FSS
+ * 0x0C - SSI0 CLK
+ * 0x0D - I2C SDA
+ * 0x0E - I2C SCL
+ * 0x0F - UART0 RX
+ * 0x10 - UART0 TX
+ * 0x11 - UART0 CTS
+ * 0x12 - UART0 RTS
+ * 0x17 - Port event 0
+ * 0x18 - Port event 1
+ * 0x19 - Port event 2
+ * 0x1A - Port event 3
+ * 0x1B - Port event 4
+ * 0x1C - Port event 5
+ * 0x1D - Port event 6
+ * 0x1E - Port event 7
+ * 0x20 - CPU SWV
+ * 0x21 - SSI1 RX
+ * 0x22 - SSI1 TX
+ * 0x23 - SSI1 FSS
+ * 0x24 - SSI1 CLK
+ * 0x25 - I2S data 0
+ * 0x26 - I2S data 1
+ * 0x27 - I2S WCLK
+ * 0x28 - I2S BCLK
+ * 0x29 - I2S MCLK
+ * 0x2E - RF Core Trace
+ * 0x2F - RF Core data out 0
+ * 0x30 - RF Core data out 1
+ * 0x31 - RF Core data out 2
+ * 0x32 - RF Core data out 3
+ * 0x33 - RF Core data in 0
+ * 0x34 - RF Core data in 1
+ * 0x35 - RF Core SMI data link out
+ * 0x36 - RF Core SMI data link in
+ * 0x37 - RF Core SMI command link out
+ * 0x38 - RF Core SMI command link in
+ */
+
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <toolchain.h>
+#include <device.h>
+#include <pinmux.h>
+#include <soc.h>
+
+
+#define IOCFG_REG(Func) \
+	REG_ADDR(TI_CC2650_PINMUX_40081000_BASE_ADDRESS, \
+		 CC2650_IOC_IOCFG0 + 0x4 * Func)
+
+static int pinmux_cc2650_init(struct device *dev);
```

Comment:
you shouldn't need these prototypes, re-order the code in the file if needed so they can be removed.